integ-tests/src/cucumberTest/java/cucumber/steps/wrappers/MemoizedWrapper.java

```diff
@@ -12,32 +12,37 @@
  * express or implied. See the License for the specific language governing
  * permissions and limitations under the License.
  */
-package steps.wrappers;
+package cucumber.steps.wrappers;
 
 import com.google.common.collect.ArrayListMultimap;
 import com.google.common.collect.ListMultimap;
 import com.google.common.collect.Multimaps;
-import java.util.List;
-import java.util.function.Function;
-import steps.helpers.Memoized;
+import org.apache.commons.lang3.Validate;
+import cucumber.steps.helpers.Memoized;
+import cucumber.steps.helpers.ThrowingFunction;
 
 public class MemoizedWrapper implements Memoized {
 
   private final ListMultimap<Class<?>, Object> memory =
       Multimaps.synchronizedListMultimap(ArrayListMultimap.create());
 
-  @Override
-  public <T> T getFromHistory(Class<T> type, Function<List<Object>, Object> fn) {
-    return null;
-  }
-
   @Override
   public <T> T getLastFromHistory(Class<T> type) {
-    return null;
+    return (T) memory.get(type).get(memory.size() - 1);
```

Comment:
Should this line be memory.get(type).get(memory.get(type).size() - 1) ?
the memory.size() will be the number of all entries for all keys.

Reply:
you're right! great catch